File: docs/WidgetMill/flutter/sdui_flutter/lib/src/builders/gesture_detector.dart
import 'package:flutter/widgets.dart';
import '../model/sdui_node.dart';
import '../renderer.dart';

/// GestureDetector — spec §2.14. Liga onTap/onLongPress/onDoubleTap ao dispatcher.
Widget buildGestureDetector(BuildContext context, SduiNode node, SduiRenderer r) {
  final e = node.events;
  VoidCallback? wire(String event) =>
      e[event] == null ? null : () => r.dispatch(e[event]);

  return GestureDetector(
    onTap: wire('onTap'),
    onLongPress: wire('onLongPress'),
    onDoubleTap: wire('onDoubleTap'),
    child: r.maybeRender(context, node.child),
  );
}

File: docs/WidgetMill/packages/spec/src/nodes/gestureDetector.ts
import { z } from "zod";
import { Events } from "../actions/index";
import { NodeRef } from "../node-ref";

/** GestureDetector — spec §2.14. Eventos: onTap, onLongPress, onDoubleTap. */
export const GestureDetectorNode = z.object({
  type: z.literal("gestureDetector"),
  events: Events.optional(),
  child: NodeRef.optional(),
});
